File: packages/shared/src/series.ts
import { z } from "zod";

export const seriesOutSchema = z.object({
    id: z.string().uuid(),
    workspace_id: z.string().uuid(),
    project_id: z.string().uuid().nullable().optional(),
    title: z.string().min(1).max(255),
    rrule_template: z.string().max(255).nullable().optional(),
    created_at: z.string().datetime(),
    updated_at: z.string().datetime(),
});

export type SeriesOut = z.infer<typeof seriesOutSchema>;

export const seriesCreateRequestSchema = z.object({
    title: z.string().min(1).max(255),
    project_id: z.string().uuid().nullable().optional(),
    rrule_template: z.string().max(255).nullable().optional(),
});

export type SeriesCreateRequest = z.infer<typeof seriesCreateRequestSchema>;

export const seriesUpdateRequestSchema = z.object({
    title: z.string().min(1).max(255).optional(),
    project_id: z.string().uuid().nullable().optional(),
    rrule_template: z.string().max(255).nullable().optional(),
});

export type SeriesUpdateRequest = z.infer<typeof seriesUpdateRequestSchema>;

File: apps/api/src/schemas/series.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class SeriesBase(BaseModel):
    title: str = Field(..., max_length=255)
    project_id: uuid.UUID | None = None
    rrule_template: str | None = Field(None, max_length=255)


class SeriesCreateRequest(SeriesBase):
    pass


class SeriesUpdateRequest(BaseModel):
    title: str | None = Field(None, max_length=255)
    project_id: uuid.UUID | None = None
    rrule_template: str | None = Field(None, max_length=255)


class SeriesOut(SeriesBase):
    id: uuid.UUID
    workspace_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
